source/AI.cpp

```diff
@@ -3006,10 +3011,15 @@ void AI::AutoFire(const Ship &ship, FireCommand &command, bool secondary) const
 	if(ship.IsYours())
 	{
 		auto it = orders.find(&ship);
-		if(it != orders.end() && it->second.target.lock() == currentTarget)
+		if(it != orders.end())
 		{
-			disabledOverride = (it->second.type == Orders::FINISH_OFF);
-			friendlyOverride = disabledOverride | (it->second.type == Orders::ATTACK);
+			if(it->second.target.lock() == currentTarget)
+			{
+				disabledOverride = (it->second.type == Orders::FINISH_OFF);
+				friendlyOverride = disabledOverride | (it->second.type == Orders::ATTACK);
+			}
+			if(it->second.type == Orders::HOLD_FIRE)
```

Comment:
If we use this as a return it should be before the lines just above
That said, I think that an order to fire should overwrite the default hold fire order, just for this target
So just make it an else if?